tiledb/sm/query/readers/result_tile.h

```diff
@@ -85,13 +85,53 @@ class ResultTile {
     TileTuple() = delete;
 
     /** Constructor with var_size and nullable parameters. */
-    TileTuple(bool var_size, bool nullable) {
+    TileTuple(
```

Comment:
I'd like to see this constructor after another refactoring pass. IMO there are too many arguments and comingled concepts. Please consider writing down the separate concepts here and give them their own names. We have:
- `{in-memory size, persisted size}` -> make struct(s)
- is it var size or nullable? -> the structs for var-size and validity can each be optional, eliminating the `bool` parameters

Something like this, as a sketch:
```
struct live_and_persisted_sizes {
  storage_size_t live,
  storage_size_t persisted
}
using data_size_pair = live_and_persisted_sizes;
using offsets_size_pair = live_and_persisted_sizes;
using validity_size_pair = live_and_persisted_sizes;

TileTuple(
  format_version,
  array_schema,
  name,
  data_size_pair,
  optional<offsets_size_pair>,
  optional<validity_size_pair>
)
```